src/store/setToolMode.js

```diff
@@ -125,17 +216,104 @@ function setToolModeForElement (mode, changeEvent, element, toolName, options) {
 /**
  * A helper/quick way to set a tool's mode for all canvases
  *
- * @param {*} mode
- * @param {*} changeEvent
- * @param {*} toolName
- * @param {*} options
+ * @param {string} mode
+ * @param {string} changeEvent
+ * @param {string} toolName
+ * @param {(object|number)} options
+ * @param {boolean} isTouchActive
  */
-function setToolMode (mode, changeEvent, toolName, options) {
+function setToolMode (mode, changeEvent, toolName, options, isTouchActive) {
   state.enabledElements.forEach((element) => {
-    setToolModeForElement(mode, changeEvent, element, toolName, options);
+    setToolModeForElement(
+      mode,
+      changeEvent,
+      element,
+      toolName,
+      options,
+      isTouchActive
+    );
   });
 }
 
+/**
+ *
+ *
+ * @param {*} element
+ * @param {*} options
+ * @param {*} isTouchActive
+ * @private
+ */
+function _setActiveToolWithMatchingMouseButtonMaskToPassive (
+  tool,
+  element,
+  options
+) {
+  const mouseButtonMask =
+    (typeof options === 'number' ? options : options.mouseButtonMask) ||
+    tool.options.mouseButtonMask;
+  const hasMouseButtonMask =
+    mouseButtonMask !== undefined && mouseButtonMask > 0;
+  const activeToolWithMatchingMouseButtonMask = state.tools.find(
+    (tool) =>
+      tool.element === element &&
+      tool.mode === 'active' &&
+      tool.options.mouseButtonMask === mouseButtonMask
+  );
+
+  if (hasMouseButtonMask && activeToolWithMatchingMouseButtonMask) {
+    console.info(
+      `Setting tool ${activeToolWithMatchingMouseButtonMask.name} to passive`
+    );
+    setToolPassiveForElement(
+      element,
+      activeToolWithMatchingMouseButtonMask.name
+    );
+  }
+}
+
+/**
+ *
+ *
+ * @param {*} element
+ * @param {*} options
+ * @param {*} isTouchActive
+ * @private
+ */
+function _setActiveToolWithIsTouchActiveToFalse (
+  tool,
+  element,
+  options,
+  isTouchActive
+) {
+  let value;
+
+  if (typeof options === 'number' && typeof isTouchActive === 'boolean') {
+    value = isTouchActive === true;
```

Comment:
Can't this just be `value = isTouchActive;`? You are already checking if its `boolean`, so its not going to be `undefined`.